snapcraft/cli/lifecycle.py

```diff
@@ -70,6 +70,13 @@ def _execute(  # noqa: C901
 
     is_managed_host = build_provider == "managed-host"
 
+    # Temporary fix to ignore target_arch.
+    if "target_arch" in kwargs and build_provider in ["multipass", "lxd"]:
+        echo.warning(
+            "Ignoring '--target-arch' flag.  This flag requires --destructive-mode, is unsupported, and will be removed in future versions."
```

Comment:
I think this might need some untangling. Maybe instead of "unsupported, "supported as-is with --destructive-mode" and a no-op on other modes or something like that

Reply:
How about `Ignoring '--target-arch' flag.  This flag requires --destructive-mode and is unsupported on LXD and Multipass`?

Reply:
updated